python/make_variable.h

```diff
@@ -135,19 +125,49 @@ Variable doMakeVariable(const std::vector<Dim> &labels, py::array &values,
     }
   }
 
-  return core::CallDType<double, float, int64_t, int32_t,
-                         bool>::apply<MakeVariable>(dtypeTag, labels, values,
+  if (dtypeTag == scipp::dtype<core::time_point>) {
+    if (variances.has_value()) {
+      throw except::VariancesError("datetimes cannot have variances.");
+    }
+    const auto [actual_unit, value_factor] = get_time_unit(values, dtype, unit);
+    if (value_factor != 1) {
+      throw std::invalid_argument(
+          "Scaling datetimes is not supported. The units of the datetime64 "
+          "objects must match the unit of the Variables.");
+    }
+    unit = actual_unit;
+  }
+
+  return core::CallDType<
+      double, float, int64_t, int32_t, bool,
+      scipp::core::time_point>::apply<MakeVariable>(dtypeTag, labels, values,
                                                     variances, unit);
 }
 
 Variable makeVariableDefaultInit(const std::vector<Dim> &labels,
                                  const std::vector<scipp::index> &shape,
-                                 const units::Unit unit, py::object &dtype,
+                                 units::Unit unit, const py::object &dtype,
                                  const bool variances) {
+  const auto dtypeTag = scipp_dtype(dtype);
+  if (dtypeTag == scipp::dtype<core::time_point>) {
+    if (variances) {
+      throw except::VariancesError("datetimes cannot have variances.");
+    }
+    const auto [actual_unit, value_factor] = get_time_unit(
+        std::nullopt,
+        dtype.is_none() ? std::optional<units::Unit>{}
+                        : parse_datetime_dtype(py::dtype::from_args(dtype)),
+        unit);
+    if (value_factor != 1) {
+      throw std::invalid_argument(
+          "Scaling datetimes is not supported. The units of the datetime64 "
+          "objects must match the unit of the Variables.");
+    }
```

Comment:
Can this exception be moved into `get_time_unit`? It is duplicated from above, and we'd also have just a single return value.

Reply:
My idea was to eventually implement this for keeping in line with how numpy does it.